File: 1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
        halve1 = s[:len(s) // 2]
        halve2 = s[len(s) // 2:]
        leftside = 0
        rightside = 0

        for i in range(len(halve1)):
            if halve1[i] in vowels:
                leftside += 1
            if halve2[i] in vowels:
                rightside += 1

        return leftside == rightside

File: 1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.ts
function halvesAreAlike(s: string): boolean {
    let vowels: string[] = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
    let halve1: string = s.slice(0, s.length / 2)
    let halve2: string = s.slice(s.length / 2, s.length)
    let leftside: number = 0
    let rightside: number = 0
for (let i = 0; i < halve1.length; i++) {
    if (vowels.includes(halve1[i])) {
        leftside++
    }
    if (vowels.includes(halve2[i])) {
        rightside++
    }
}

    return leftside==rightside
};